packages/synapse-react-client/src/components/QueryContext/QueryContext.tsx

```diff
@@ -81,6 +82,18 @@ export type QueryContextType<
   getColumnModel: (columnName: string) => ColumnModel | null
   // Either open benefactor entity page in a new window or open the sharing settings dialog (in Synapse.org)
   onViewSharingSettingsClicked?: (benefactorId: string) => void
+  /** Whether the user can select individual rows on which to perform an action */
+  isRowSelectionVisible: boolean
+  /** The collection of selected rows */
+  selectedRows: Row[]
+  /** State updater for `selectedRows`  */
+  setSelectedRows: React.Dispatch<React.SetStateAction<Row[]>>
+  /** The set of columns that defines a uniqueness constraint on the table for the purposes of filtering based on row selection.
+   * Note that Synapse tables have no internal concept of a primary key.
+   */
+  rowSelectionPrimaryKey?: string[]
+  /** Whether the user has selected any rows */
+  hasSelectedRows: boolean
```

Comment:
Move row selection to `QueryContext` since `QueryWrapper` needs to know about row selection in order to conditionally block changing the query.